sound/soc/sof/intel/hda.h

```diff
@@ -648,6 +648,11 @@ void hda_dsp_ctrl_stop_chip(struct snd_sof_dev *sdev);
 void sof_hda_bus_init(struct hdac_bus *bus, struct device *dev);
 
 #if IS_ENABLED(CONFIG_SND_SOC_SOF_HDA)
+/*
+ * HDA Codec module_param.
+ */
+extern int hda_codec_mask;
+
```

Comment:
Please export a getter function instead of exporting the variable directly.

Reply:
I checked other usage about `module_param` and find this kind of extern usage to share param to other files in same module. @plbossart Any suggestion here, I think use the original module param will be more clear to show we are overriding by module param here.

Reply:
I will let @kv2019i chime in on how he wants the HDaudio parameters to be handled.

Reply:
Which drivers @xiulipan you used as reference? 

It seems hard to find any examples where module params are exported to other compilation units. Of the few ones I found, e.g. drivers/edac/edac_mc_sysfs.c exports getter functions. To me this looks safer than allowing raw access to data across modules.

Not a big issue, but given this is somewhat unusual usage, even more important to make the code easy to understand. 

Rest of review will come later, didn't have time today.

Reply:
Sure, I just do a quick search in Linux folder, not remember which. But if you also find some usage like getter, I will try to update it.

Reply:
I tried to add the getter functions, but I get into some `Symbol Namespaces` issue.

`hda-codec.c` is in `snd_sof_intel_hda`
`hda-ctrl.c` is in `snd_sof_intel_hda_common`

and if I want to use some function in `hda-ctrl.c` from `hda-codec.c` I will get error:

`depmod: ERROR: Cycle detected: snd_sof_intel_hda_common -> snd_sof_intel_hda -> snd_sof_intel_hda_common`
@kv2019i @plbossart I am thinking move the whole module_param and getter into `snd_sof_intel_hda` so we won't have above issues. Any idea? Please help check if we will get conflict with https://github.com/thesofproject/linux/pull/2683

Reply:
Hmm, that's a problem @xiulipan indeed. Moving the kernel parameter is problematic as that can break existing setups (users have used the old parameter in their modprobe.d conf files and if we change the param to a different module, this will break). I'll also investigate whether there's a cleaner way.

Reply:
@kv2019i the module parameter `hda_codec_mask` is in module `snd_sof_intel_hda_common`
so maybe we kept this variable usage?